oak_functions/abi/proto/abi.proto

```diff
@@ -0,0 +1,32 @@
+//
+// Copyright 2021 The Project Oak Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+
+syntax = "proto3";
+
+package oak_functions_abi;
```

Comment:
We already have https://github.com/project-oak/oak/tree/main/oak_functions/proto which

- is in a different folder
- has a different package name
- is built as part of a different crate

It seems we should just unify all these things?

Reply:
Yes. I was not sure where is the right place for this proto file. The `oak_abi` crate has its own `proto` folder. Why is that one not merged with the top `proto` folder?

Reply:
I think it's just historical reasons. For Oak Functions, I think we can use either `oak_functions/proto` or `oak_functions/abi/proto`, perhaps the first one is better since it's not strictly speaking just part of the ABI? OTOH you could consider the key value stuff as part of the ABI too in a way... not a big deal anyways, we can easily move them around later.

Reply:
Moved it to `oak_functions/proto`, since there is no good reason to have a separate `proto` file.